shop-frontend-angular-redux/src/app/products/api/actions.ts

```diff
@@ -0,0 +1,76 @@
+import {ProductRestResponse, ProductCategoryRestResponse, Product, ProductCategory} from "../products.types";
+import {Injectable} from '@angular/core';
+import {dispatch} from '@angular-redux/store';
+import {FluxStandardAction} from 'flux-standard-action';
+
+export type ProductAPIAction = FluxStandardAction<Product[], any>;
+export type ProductCategoryAPIAction = FluxStandardAction<ProductCategory[], any>;
+
+@Injectable()
+export class ProductActions {
+    static readonly LOAD_PRODUCTS = "LOAD_PRODUCTS";
+    static readonly LOAD_PRODUCTS_STARTED = 'LOAD_PRODUCTS_START';
+    static readonly LOAD_PRODUCTS_SUCCEEDED = 'LOAD_PRODUCTS_SUCCEEDED';
+    static readonly LOAD_PRODUCTS_FAILED = 'LOAD_PRODUCTS_FAILED';
+
+    @dispatch()
+    loadProducts = (): ProductAPIAction => ({
+        type: ProductActions.LOAD_PRODUCTS,
+        meta: null,
```

Comment:
Das gleiche für Payload, wenn nichts übergeben wird. Ich würde in dem Fall das Feld ebenfalls weglassen. Das müsste vom TS-Compiler eigentlich akzeptiert werden